Fix: Poor contrast in cargo dest flow legend window cargo labels

JGRennison opened the pull request:
Select foreground colour depending on the brightness of the background.

Previously all cargo labels were rendered using black text, even
the background cargo colour was dark/black.
As an example: FIRS coal was black text on a black background.

LordAro commented on src/gfx.cpp:
```diff
@@ -1113,14 +1113,14 @@ void DoPaletteAnimations()
  * @return TC_BLACK or TC_WHITE depending on what gives a better contrast.
  */
-TextColour GetContrastColour(uint8 background)
+TextColour GetContrastColour(uint8 background, uint8 threshold)
```
doc comments should be extended accordingly

JGRennison replied:
Thanks, this should be fixed in the most recent update of the branch.

JGRennison committed:
Fix: Poor contrast in cargo dest flow legend window cargo labels.

Select foreground colour depending on the brightness of the background.

Previously all cargo labels were rendered using black text, even
the background cargo colour was dark/black.
As an example: FIRS coal was black text on a black background.
